### PR TITLE
Add autotile tutorial

### PR DESCRIPTION
**Pull request by PeterJSFChaplin**
Modify tutorials > 2D > using tilemaps to include section on
autotiles, covering different bitmask modes and scripted binding

### CODE REVIEW

File: tutorials/2d/using_tilemaps.rst [hunk 1]
```diff
@@ -150,6 +152,108 @@ tiles it contains:
 In addition to saving time when defining the tiles, this can help by grouping
 similar tiles together when you're working with a large number of tiles.
 
+Random tile priorities
+~~~~~~~~~~~~~~~~~~~~~~
+
+When drawing with atlas tiles, enabling the "Use priority" option causes tiles
+to be selected at random. By default, each tile in the tileset has an equal
+likelihood of occuring. You can change the likelihood by setting different
```
Comment (mhilbrunner) [inline tutorials/2d/using_tilemaps.rst:160]:
```suggestion
likelihood of occurring. You can change the likelihood by setting different
```

File: tutorials/2d/using_tilemaps.rst [hunk 2]
```diff
@@ -150,6 +152,108 @@ tiles it contains:
...
+---------
+
+Autotiles allow you to define a group of tiles, then add rules to control which
+tile gets used for drawing based on the content of adjacent cells.
+
+Click "New Autotile" and drag to select the tiles you wish to use. You can add
+collisions, occlusion, navigation shapes, tile priorties, and select an icon
+tile in the same manner as for atlas tiles.
+
+Tile selection is controled by bitmasks. Bitmasks can be added by clicking
```
Comment (mhilbrunner) [inline tutorials/2d/using_tilemaps.rst:175]:
```suggestion
Tile selection is controlled by bitmasks. Bitmasks can be added by clicking
```

**Comment by mhilbrunner** [comment]
Fixed two small typos. Otherwise looks good to me :)

**Comment by cbscribe** [comment]
I think it would be very helpful to include a mention of how many tiles you need in your tileset for each bitmask mode. This question comes up a lot.

**Comment by PeterJSFChaplin** [comment]
@cbscribe I'm a little wary of how we word that - technically you can create a tileset with only two subtiles for any bitmask mode, if there's some reason you _want_ an autotile with only two subtiles. Especially with the addition of wildcards (aka "ignore" bits).

It depends on what behaviour you want and/or expect.

**Comment by PeterJSFChaplin** [comment]
Ok, I've tried to include some text about the number of supported bitmasks in each mode

**Comment by cbscribe** [comment]
I hear you, but in the support channels autotile is a constant source of confusion. I'm still trying to come up with a better way to unambiguously demonstrate how it works. At least with a fixed, "here are the 47 tiles you need for 3x3 (minimal)" example, people have a place to start.

Thanks - I think this is good to go.

**Comment by mhilbrunner** [comment]
Thanks - and congrats on your first contribution! :)

**Pull request merged**